C++14/Maths/Basics/FastDoubling_Method.cpp

```diff
@@ -0,0 +1,42 @@
+#include <bits/stdc++.h>
+using namespace std;
+#define ll long long int
+#define MOD 1000000007;
+ll a,b,c,d;

```

Comment:
Avoid global variables and use local variables inside the function.

Reply:
Use `int` since the mod value fit's inside it. 

Check https://github.com/likecs/Competitive-Coding/blob/master/C%2B%2B14/Maths/Basics/fibonacci_mod_m.cpp for example where optimal calculations are being used using helper functions like `add`, `mul` etc.

Reply:
Ok...I will see through the guidelines and commit the changes accordingly